### PR TITLE
Cypress: Initial implementation of MCUBoot for 062 - BSP less

### PR DESCRIPTION
**Pull request by romanjoe**
This PR is created to substitute existing one https://github.com/JuulLabs-OSS/mcuboot/pull/605.

This implementation is not dependent on BSP asset, as previous one. All the dependencies used as submodules are publicly available.

Features:

- All the code is contained in boot/cypress folder and do not change any MCUBoot source code
- Makefile based build support implemented
- Dependencies are provided as submodules
- image signature and hash validation implemented using software mbedtls
- BlinkyApp - test application provided to quickly test MCUBootApp which is a bootloader.
- Information regarding bootloader and test application build can be found in corresponding Readme.md files.

### CODE REVIEW

**Comment by utzig** [comment]
Please rebase on `master`.

**Comment by utzig** [comment]
@romanjoe This still adds a `mbedtls` submodule from cypress, can't we use the one we already have at `ext/mbedtls`?

**Comment by romanjoe** [comment]
> @romanjoe This still adds a `mbedtls` submodule from cypress, can't we use the one we already have at `ext/mbedtls`?

Looks like it is the same instance, as we have - from base https://github.com/ARMmbed/mbedtls, so i suppose we can substitute. When we first started this work, there were some subset of mbedtls in upstream, that is why we added full repo to our folders.

File: boot/cypress/MCUBootApp/image_ec256_mbedtls.c [hunk 1]
```diff
@@ -0,0 +1,166 @@
+/*
```
Comment (utzig) [inline boot/cypress/MCUBootApp/image_ec256_mbedtls.c]: This file could probably be refactored to go into `boot/bootutil/src/image_ec256.c`.

**Comment by romanjoe** [comment]
- fixed code per Fabio comments
- rebased this branch against Fabios PR with MbedTLS version update
- got rid of cypress instance of MbedTLS, using ext/mbedtls now
- updated makefiles accordingly

File: boot/cypress/BlinkyApp/BlinkyApp.mk [hunk 2]
```diff
@@ -14,7 +14,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+#	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (utzig) [inline boot/cypress/BlinkyApp/BlinkyApp.mk]: This should be a space

File: boot/cypress/BlinkyApp/BlinkyApp.mk [hunk 3]
```diff
@@ -25,9 +25,9 @@
 
 # Cypress' MCUBoot Application supports GCC ARM only at this moment
 # Set defaults to:
-# 	- compiler GCC
-#   - build configuration to Debug
-#   - image type to BOOT
+#	- compiler GCC
+#	- build configuration to Debug
+#	- image type to BOOT
```
Comment (utzig) [inline boot/cypress/BlinkyApp/BlinkyApp.mk]: These would align better if they were spaces

File: boot/cypress/BlinkyApp/libs.mk [hunk 4]
```diff
@@ -14,7 +14,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+#	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (utzig) [inline boot/cypress/BlinkyApp/libs.mk]: tab -> space

File: boot/cypress/BlinkyApp/platforms.mk [hunk 5]
```diff
@@ -14,7 +14,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+#	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (utzig) [inline boot/cypress/BlinkyApp/platforms.mk]: tab -> space

File: boot/cypress/BlinkyApp/toolchains.mk [hunk 6]
```diff
@@ -14,7 +14,7 @@
 # you may not use this file except in compliance with the License.
 # You may obtain a copy of the License at
 #
-#     http://www.apache.org/licenses/LICENSE-2.0
+#	http://www.apache.org/licenses/LICENSE-2.0
```
Comment (utzig) [inline boot/cypress/BlinkyApp/toolchains.mk]: tab -> space

File: boot/cypress/BlinkyApp/toolchains.mk [hunk 7]
```diff
@@ -24,10 +24,10 @@
...
 
 # Compilers
-GCC_ARM   	:= 1
-IAR   		:= 2
-ARM   		:= 3
-OTHER 		:= 4
+GCC_ARM	:= 1
+IAR		:= 2
+ARM		:= 3
+OTHER 	:= 4
```
Comment (utzig) [inline boot/cypress/BlinkyApp/toolchains.mk]: Would align better if was using spaces here

**Comment by d3zd3z** [comment]
I think this looks ok, as most of the change is within the Cypress directory. I'll be curious if the extra submodules end up being challenging for people on other targets.

I still think we should try to get Cypress support through the various RTOS's. Effectively, this is adding another RTOS, that just happens to be a lot thinner. But, this isn't really going to scale for adding more targets this way.

**Comment by utzig** [comment]
Could you please squash f0e95a02e7d8e50f65c41d382cb97ebb9bd82f2a, 0a4a53c9c9de9d119ae8f9ea38a4a5739919f507 and 8a44b8d275702e5e64e9e41bebd13bc072288525?

**Pull request merged**